.github/workflows/ci.yml

```diff
@@ -0,0 +1,73 @@
+name: Continuous Integration
+
+on: [pull_request, workflow_dispatch]
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  ci:
+    name: CI
+    runs-on: ubuntu-latest
+    if: |
+      github.event_name == 'push' ||
+      (github.event_name == 'pull_request' && github.event.pull_request.draft == false)
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.8']
+        torch-version: ['1.11', '2.2']
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        lfs: false
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Cache dependency
+      id: cache-dependencies
+      uses: actions/cache@v4
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install torch==${{ matrix.torch-version }} --extra-index-url https://download.pytorch.org/whl/cpu
+        pip install -e .[dev]
+
+    - name: Install and run pre-commit hooks
```

Comment:
as I commented above. if we are using the pre-commit action, we can run it without installing our dependencies, which saves a lot of time (pre-commit actions runs within 10s after pushing).

Reply:
ok, well I guess this now really reads like I should change it back :rofl:

Reply:
my comment above was unclear, I admit.